File: CicloDeVida/app/src/main/java/com/example/ciclodevida/MainActivity.java
package com.example.ciclodevida;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d("Ciclo de Vida","onCreate");
    }


    @Override
    protected void onStart() {
        super.onStart();
        Log.d("Ciclo de Vida","onCreate");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d("Ciclo de Vida","onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d("Ciclo de Vida","onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d("Ciclo de Vida","onStop");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d("Ciclo de Vida","onRestart");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d("Ciclo de Vida","onDestroy");
    }
}

File: dados/app/src/main/java/br/com/mariojp/dados/MainActivity.kt
package br.com.mariojp.dados

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import br.com.mariojp.dados.databinding.ActivityMainBinding
import java.util.Random

class MainActivity : AppCompatActivity() {

    //lateinit var resultado : ImageView;
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main)
        //val botao: Button = findViewById(R.id.main_button_lancar)
        //botao.text = "Jogar"
        binding.mainButtonLancar.text = "Jogar"
        binding.mainButtonLancar.setOnClickListener {
                jogar()
            }
        //resultado = findViewById(R.id.main_image_dado)

    }


    private fun jogar(){
        val numero = Random().nextInt(6) + 1
        val drawableResource = when (numero) {
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }
        //resultado.setImageResource(drawableResource);
        binding.mainImageDado.setImageResource(drawableResource);
    }
}